packages/typespec-python/src/emitter.ts

```diff
@@ -538,12 +538,24 @@ function emitResponse(
     context: SdkContext,
     response: HttpOperationResponse,
     innerResponse: HttpOperationResponseContent,
+    operation: Operation,
 ): Record<string, any> {
     let type = undefined;
+    let resultProperty = undefined;
     if (innerResponse.body?.type) {
         let modelType = undefined;
         if (innerResponse.body.type.kind === "Model") {
             modelType = getEffectiveSchemaType(context, innerResponse.body.type);
+            // For LROs, we use getLroMetadata to get final result type if we can't get it from common function
+            if (isAzureCoreModel(modelType) && modelType.name === "ResourceOperationStatus") {
```

Comment:
Is there more common way that we could use to judge the actual resource is inside the status model? This comparasion is a little bit hacky.

Reply:
Yeah, the logic is hacky. Currently I can't find proper function or better logic to judge it.

Reply:
`lroOperation.response = getType(context, logicalResult.type)`

Reply:
update the logic to handle all possible cases.